File: frontend/libs/portal/model/InternalToken.ts
// SPDX-FileCopyrightText: 2025 Mercedes-Benz Group AG and Mercedes-Benz AG
//
// SPDX-License-Identifier: Apache-2.0

export const capabilityMap = {
  StatisticsCSV: 0,
  CustomLicenses: 1,
};
export type Capability = keyof typeof capabilityMap;

export interface InternalToken {
  _key?: string;
  updated?: string;
  created?: string;
  name: string;
  revoked: boolean;
  description: string;
  expiry: string;
  token: string;
  capabilities: Capability[];
}

// Shared interface for create/update operations
export interface InternalTokenRequest {
  _key?: string;
  name: string;
  revoked?: boolean;
  token?: string;
  description: string;
  expiry: string;
  capabilities: Capability[];
}

File: backend/domain/internalToken/internaltoken.go
// SPDX-FileCopyrightText: 2025 Mercedes-Benz Group AG and Mercedes-Benz AG
//
// SPDX-License-Identifier: Apache-2.0

package internalToken

import (
	"time"

	"mercedes-benz.ghe.com/foss/disuko/domain"
)

type Capability int

const (
	StatisticsCSV Capability = iota
	CustomLicenses
)

type InternalToken struct {
	domain.RootEntity `bson:"inline"`
	Name              string
	Revoked           bool
	Token             string
	Salt              string
	Description       string
	Expiry            time.Time
	Capabilities      []Capability
}
